Added documentation for ocean_aks_2.0 cluster and vng

sharadkesarwani opened the pull request:
Added documentation for ocean_aks_2.0 cluster and vng.

# Jira Ticket
Ref:https://spotinst.atlassian.net/browse/SPOTAUT-15257

sharadkesarwani committed:
Added documentation for ocean_aks_2.0 cluster and vng

Added documentation for ocean_aks_2.0 cluster and vng

sharadkesarwani committed:
Updated field description

Updated field description

sharadkesarwani committed:
updated page title

updated page title

chandra1-n commented on docs/resources/ocean_aks_2.0.md:
```diff
@@ -0,0 +1,244 @@
+
+  # Configuration.
+  cluster_identifier = "ocean-westus-dev-aks"
+  acd_identifier     = "acd-12345678"
```
check this param if it's required

sharadkesarwani replied:
removed acd_identifier for now

chandra1-n commented on docs/resources/ocean_aks_2.0.md:
````diff
@@ -0,0 +1,244 @@
+}
+```
+
+~> You must configure the same `cluster_identifier` and `acd_identifier` both for the Ocean controller and for the `spotinst_ocean_aks` resource.
````
check for acd_identifier param

sharadkesarwani replied:
removed acd_identifier for now

chandra1-n commented on docs/resources/ocean_aks_2.0.md:
```diff
@@ -0,0 +1,244 @@
+* `scheduling` - (Optional) An object used to specify times when the cluster will turn off. Once the shutdown time will be over, the cluster will return to its previous state.
+    * `shutdown_hours` - (Optional) An object used to specify times that the nodes in the cluster will be taken down.
+        * `is_enabled` - (Optional) Flag to enable or disable the shutdown hours mechanism. When False, the mechanism is deactivated, and the cluster remains in its current state.
+        * `time_windows` - (Optional) The times that the shutdown hours will apply.
```
remove optional and mention in braces that "required if is_enabled set to true"

chandra1-n commented on docs/resources/ocean_aks_2.0.md:
```diff
@@ -0,0 +1,244 @@
+        * `time_windows` - (Optional) The times that the shutdown hours will apply.
+* `headrooms` - (Optional) Specify the custom headroom per VNG. Provide a list of headroom objects.
+  * `cpu_per_unit` - (Optional) Configure the number of CPUs to allocate the headroom. CPUs are denoted in millicores, where 1000 millicores = 1 vCPU.
+  * `memory_per_unit` - (Optional) Amont of GPU to allocate for headroom unit.
```
fix the description which is inter-changed

sharadkesarwani replied:
updated descriptions

chandra1-n commented on docs/resources/ocean_aks_2.0.md:
```diff
@@ -0,0 +1,244 @@
+  * `memory_per_unit` - (Optional) Amont of GPU to allocate for headroom unit.
+  * `gpu_per_unit` - (Optional) Configure the amount of memory (MiB) to allocate the headroom.
+  * `num_of_units` - (Optional) The number of units to retain as headroom, where each unit has the defined headroom CPU and memory.
+* `availability_zones` - (Optional) An Array holding Availability Zones, this configures the availability zones the Ocean may launch instances in per VNG.
```
it is required param, need to be fixed in code

chandra1-n commented on docs/resources/ocean_aks_2.0.md:
```diff
@@ -0,0 +1,244 @@
+  * `gpu_per_unit` - (Optional) Configure the amount of memory (MiB) to allocate the headroom.
+  * `num_of_units` - (Optional) The number of units to retain as headroom, where each unit has the defined headroom CPU and memory.
+* `availability_zones` - (Optional) An Array holding Availability Zones, this configures the availability zones the Ocean may launch instances in per VNG.
+* `labels` - (Optional) An array of labels to add to the virtual node group.Only custom user labels are allowed, and not Kubernetes built-in labels or Spot internal labels.
```
node group.Only give space in-between

sharadkesarwani replied:
done

chandra1-n commented on docs/resources/ocean_aks_2.0.md:
```diff
@@ -0,0 +1,244 @@
+* `availability_zones` - (Optional) An Array holding Availability Zones, this configures the availability zones the Ocean may launch instances in per VNG.
+* `labels` - (Optional) An array of labels to add to the virtual node group.Only custom user labels are allowed, and not Kubernetes built-in labels or Spot internal labels.
+    * `key` - (Required) Set label key.
+    * The following are not allowed:
```
this shouldn't be bullet points

sharadkesarwani replied:
made it list[string]

chandra1-n commented on docs/resources/ocean_aks_2.0.md:
```diff
@@ -0,0 +1,244 @@
+* `labels` - (Optional) An array of labels to add to the virtual node group.Only custom user labels are allowed, and not Kubernetes built-in labels or Spot internal labels.
+    * `key` - (Required) Set label key.
+    * The following are not allowed:
+    * "kubernetes.azure.com/agentpool"
```
this shouldn't be bullet points

sharadkesarwani replied:
removed bullet points and made it to list[string]

chandra1-n commented on docs/resources/ocean_aks_2.0.md:
```diff
@@ -0,0 +1,244 @@
+    min_memory_gib = 10
+    max_memory_gib = 18
+    architectures = ["X86_64"]
+    series = ["D v3"]
```
include "Ddsv5", "Dds_v4" as well

sharadkesarwani replied:
included

chandra1-n commented on docs/resources/ocean_aks_vng_2.0.md:
```diff
@@ -0,0 +1,154 @@
+---
+layout: "spotinst"
+page_title: "Spotinst: ocean_aks_np_vng"
```
expand vng

sharadkesarwani replied:
updated to ocean_aks_np_virtual_node_group

chandra1-n commented on docs/resources/ocean_aks_vng_2.0.md:
```diff
@@ -0,0 +1,154 @@
+    min_memory_gib = 10
+    max_memory_gib = 18
+    architectures = ["X86_64"]
+    series = ["D v3"]
```
include "Ddsv5", "Dds_v4" as well

sharadkesarwani replied:
included

chandra1-n commented on docs/resources/ocean_aks_vng_2.0.md:
```diff
@@ -0,0 +1,154 @@
+  
+  // ----------------------------------------------------------------------------
+  
+  // ----------------------------------------------------------------------------
```
remove this extra line

sharadkesarwani replied:
removed

chandra1-n commented on docs/resources/ocean_aks_2.0.md:
```diff
@@ -0,0 +1,244 @@
+    * `shutdown_hours` - (Optional) An object used to specify times that the nodes in the cluster will be taken down.
+        * `is_enabled` - (Optional) Flag to enable or disable the shutdown hours mechanism. When False, the mechanism is deactivated, and the cluster remains in its current state.
+        * `time_windows` - (Optional) The times that the shutdown hours will apply.
+* `headrooms` - (Optional) Specify the custom headroom per VNG. Provide a list of headroom objects.
```
it says per VNG instead of cluster, get clarification on openAPI document then change accordingly

chandra1-n commented on docs/resources/ocean_aks_vng_2.0.md:
```diff
@@ -0,0 +1,154 @@
+* `ocean_id` - (Required) The Ocean cluster identifier. Required for Launch Spec creation.
+* `headrooms` - (Optional) Specify the custom headroom per VNG. Provide a list of headroom objects.
+  * `cpu_per_unit` - (Optional) Configure the number of CPUs to allocate the headroom. CPUs are denoted in millicores, where 1000 millicores = 1 vCPU.
+  * `memory_per_unit` - (Optional) Amont of GPU to allocate for headroom unit.
```
correct the description

sharadkesarwani replied:
corrected description

chandra1-n commented on docs/resources/ocean_aks_vng_2.0.md:
```diff
@@ -0,0 +1,154 @@
+* `headrooms` - (Optional) Specify the custom headroom per VNG. Provide a list of headroom objects.
+  * `cpu_per_unit` - (Optional) Configure the number of CPUs to allocate the headroom. CPUs are denoted in millicores, where 1000 millicores = 1 vCPU.
+  * `memory_per_unit` - (Optional) Amont of GPU to allocate for headroom unit.
+  * `gpu_per_unit` - (Optional) Configure the amount of memory (MiB) to allocate the headroom.
```
correct the description

sharadkesarwani replied:
corrected

chandra1-n commented on docs/resources/ocean_aks_vng_2.0.md:
```diff
@@ -0,0 +1,154 @@
+  * `memory_per_unit` - (Optional) Amont of GPU to allocate for headroom unit.
+  * `gpu_per_unit` - (Optional) Configure the amount of memory (MiB) to allocate the headroom.
+  * `num_of_units` - (Optional) The number of units to retain as headroom, where each unit has the defined headroom CPU and memory.
+* `availability_zones` - (Optional) An Array holding Availability Zones, this configures the availability zones the Ocean may launch instances in per VNG.
```
get clarity on required or optional

chandra1-n commented on docs/resources/ocean_aks_vng_2.0.md:
```diff
@@ -0,0 +1,154 @@
+  * `num_of_units` - (Optional) The number of units to retain as headroom, where each unit has the defined headroom CPU and memory.
+* `availability_zones` - (Optional) An Array holding Availability Zones, this configures the availability zones the Ocean may launch instances in per VNG.
+* `labels` - (Optional) An array of labels to add to the virtual node group.Only custom user labels are allowed, and not Kubernetes built-in labels or Spot internal labels.
+    * `key` - (Required) Set label key.
```
alignment is required

sharadkesarwani replied:
Aligned.

chandra1-n commented on docs/resources/ocean_aks_vng_2.0.md:
```diff
@@ -0,0 +1,154 @@
+* `os_disk_type` - (Optional) The type of the OS disk.
+* `os_type` - (Optional) The OS type of the OS disk.
+* `fallback_to_ondemand` - (Optional, Default: `true`) If no spot instance markets are available, enable Ocean to launch on-demand instances instead.
+* `spot_percentage` - (Optional,Default: `100`) Percentage of spot VMs to maintain.
```
give space in-between

sharadkesarwani replied:
gave space before default

chandra1-n reviewed (changes requested):
please fix the comments

chandra1-n commented on docs/resources/ocean_aks_2.0.md:
```diff
@@ -0,0 +1,244 @@
+* `enable_node_public_ip` - (Optional) Enable node public IP.
+* `max_pods_per_node` - (Optional) The maximum number of pods per node in the node pools.
+* `os_disk_size_gb` - (Optional) The size of the OS disk in GB.
+* `os_disk_type` - (Optional) The type of the OS disk.
```
mention Enum

chandra1-n commented on docs/resources/ocean_aks_vng_2.0.md:
```diff
@@ -0,0 +1,154 @@
+* `enable_node_public_ip` - (Optional) Enable node public IP.
+* `max_pods_per_node` - (Optional) The maximum number of pods per node in the node pools.
+* `os_disk_size_gb` - (Optional) The size of the OS disk in GB.
+* `os_disk_type` - (Optional) The type of the OS disk.
```
mention Enum

sharadkesarwani committed:
Updated sdk version and documentation related to aks 2.0 cluster and vng

Updated sdk version and documentation related to aks 2.0 cluster and vng
Updated old aks cluster and vng doc as deprecated

sharadkesarwani committed:
Updated dependencies using command

Updated dependencies using command

sharadkesarwani committed:
Updated file name

Updated file name from ocean_aks_np_vng.md to ocean_aks_np_virtual_node_group.md

sharadkesarwani committed:
Updated field values as present in openApi for cluster and vng

Updated field values as present in openApi for cluster and vng

chandra1-n commented on docs/resources/ocean_aks_np.md:
```diff
@@ -0,0 +1,224 @@
+    min_memory_gib = 8
+    max_memory_gib = 16
+    architectures = ["X86_64"]
+    series = ["D v3", "F", "E v4"]
```
add the series which we discussed earlier for both cluster and vng

chandra1-n commented on docs/resources/ocean_aks_np.md:
```diff
@@ -0,0 +1,224 @@
+  
+  // ----------------------------------------------------------------------------
+  
+  // ----------------------------------------------------------------------------
```
remove this extra line

chandra1-n reviewed (changes requested):
please fix the comments

sharadkesarwani committed:
Updated series and removed a line

Updated series and removed a line

chandra1-n committed:
Merge branch 'main' into SPOTAUT-15257-aks2.0_doc
